src/phoenix/trace/filter.py

```diff
@@ -0,0 +1,180 @@
+import ast
+from typing import Any, Iterator, Mapping, Tuple, cast
+
+from phoenix.trace import semantic_conventions
+from phoenix.trace.schemas import Span
+
+
+class Filter:
+    def __init__(self, condition: str) -> None:
+        self._root = ast.parse(condition, mode="eval")
+        _validate_expression(self._root, condition)
+        self._translated = _Translator(condition).visit(self._root)
+        ast.fix_missing_locations(self._translated)
+        self._compiled = compile(self._translated, filename="", mode="eval")
+
+    def __call__(self, span: Span) -> bool:
+        return cast(bool, eval(self._compiled, {}, {"span": span, "_MISSING": _Missing()}))
+
+
+def _replace_none_with_missing(
+    value: ast.expr,
+    as_str: bool = False,
+) -> ast.IfExp:
+    """E.g. `value` becomes
+    `_MISSING if (_MAYBE := value) is None else _MAYBE`
+    """
```

Comment:
```suggestion
    """
    E.g. `value` becomes
    `_MISSING if (_MAYBE := value) is None else _MAYBE`
    """
```

Reply:
I'm not sure I follow this function definition. Can you add more examples? I don't know what `_MAYBE` is